packages/docusaurus-plugin-content-docs/src/client/docsClientUtils.ts

```diff
@@ -25,17 +25,6 @@ export type GetActivePluginOptions = {failfast?: boolean};
 // get the data of the plugin that is currently "active"
 // ie the docs of that plugin are currently browsed
 // it is useful to support multiple docs plugin instances
-export function getActivePlugin(
-  allPluginDatas: Record<string, GlobalPluginData>,
-  pathname: string,
-  options: {failfast: true}, // use fail-fast option if you know for sure one plugin instance is active
-): ActivePlugin;
-export function getActivePlugin(
-  allPluginDatas: Record<string, GlobalPluginData>,
-  pathname: string,
-  options?: GetActivePluginOptions,
-): ActivePlugin | undefined;
```

Comment:
wonder why this is removed?

https://mariusschulz.com/blog/function-overloads-in-typescript#version-4-function-overloads

this is TS function overload to have more precise typing when some options are used (ie different return type). Does it cause any problem with ESLint?

Reply:
Ah, my bad, fixed it.